app/models/comment.rb

```diff
@@ -5,10 +5,26 @@ class Comment < ApplicationRecord
 
   belongs_to :source, polymorphic: true
   belongs_to :user
+  has_many :notifications, as: :source, dependent: :destroy
 
   validates :user_id, presence: true
   validates :body, length: { maximum: 2000 }, presence: true
 
+  # FIXME: This code assumes that submissions are the only possible thing to comment on,
+  #   and although as far as I know that is *currently* true, the source is polymorphic,
+  #   so this should be rewritten to act on *any* source_type.
+  #   I don't know how to do it personally, but I assume there are two possibilities:
+  #     1. link_form does it somehow, and that code should be abstracted out to this method
+  #     2. link_form doesn't handle it, in which case it's probably safe to ignore for now
+  def url
+    "/submissions/#{source_id}\##{id}"
```

Comment:
Change this to adapt to source_type (to set a precedent for the future) and we should be good.